roles/openshift_version/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+  when: openshift_release is defined and openshift_release[0] == 'v'
+
+- set_fact:
+    openshift_image_tag: "{{ 'v' + openshift_image_tag }}"
+  when: openshift_image_tag is defined and openshift_image_tag[0] != 'v'
+
+- set_fact:
+    openshift_pkg_version: "{{ '-' + openshift_pkg_version }}"
+  when: openshift_pkg_version is defined and openshift_pkg_version[0] != '-'
+
+# Make sure we copy this to a fact if given a var:
+- set_fact:
+    openshift_version: "{{ openshift_version }}"
+  when: openshift_version is defined
+
+# Protect the installed version by default unless explicitly told not to, or given an
+# openshift_version already.
+- name: Use openshift.common.version fact as version to configure if already installed
+  set_fact:
+    openshift_version: "{{ openshift.common.version }}"
+  when: openshift.common.version is defined and openshift_version is not defined and openshift_protect_installed_version | bool
+
+- debug: var=is_containerized
+- debug: var=openshift_version
+- debug: msg="{{ openshift_version is defined }}"
+- debug: var=openshift_release
+- debug: var=openshift_pkg_version
+- debug: var=openshift_image_tag
+- debug: var=openshift.common.version
+
+- name: Set openshift_version for rpm installation
+  include: set_version_rpm.yml
+  when: not is_containerized | bool
+
+- name: Set openshift_version for containerized installation
+  include: set_version_containerized.yml
+  when: is_containerized | bool
+
+# At this point we know openshift_version is set appropriately. Now we set
+# openshift_image_tag and openshift_pkg_version, so all roles can always assume
+# each of this variables *will* be set correctly and can use them per their
+# intended purpose.
+
+- set_fact:
+    openshift_image_tag: v{{ openshift_version }}
+  when: openshift_image_tag is not defined
+
+- set_fact:
+    openshift_pkg_version: -{{ openshift_version }}
+  when: openshift_pkg_version is not defined
+
+- debug: var=openshift_version
+- debug: var=openshift_pkg_version
+- debug: var=openshift_image_tag
+
+- fail:
+    msg: openshift_version role was unable to set openshift_version
+  when: openshift_version is not defined
+
+- fail:
+    msg: openshift_version role was unable to set openshift_image_tag
+  when: openshift_image_tag is not defined
+
+- fail:
+    msg: openshift_version role was unable to set openshift_pkg_version
+  when: openshift_pkg_version is not defined
+
+# We can't map an openshift_release to full rpm version like we can with containers, make sure
+# the rpm version we looked up matches the release requested and error out if not.
+- fail:
+    msg: "Detected openshift version {{ openshift_version }} does not match requested openshift_release {{ openshift_release }}. You may need to adjust your yum repositories or specify an exact openshift_pkg_version."
+  when: not is_containerized | bool and openshift_release is defined and not openshift_version | oo_startswith(openshift_release) | bool
```

Comment:
So confused, I spent the better part of an hour trying to get that to work and couldn't, but you are correct and it's fine now. Must have botched something else in the line. I've updated and remove oo_startswith.